paasta_tools/cli/cmds/local_run.py

```diff
@@ -519,6 +538,26 @@ def check_if_port_free(port):
     return True
 
 
+def decrypt_secret_environment_variables(
+    secret_provider_name,
+    environment,
+    soa_dir,
+    service_name,
+    cluster_name,
+    secret_provider_kwargs,
+):
+    ecosystem = secret_provider_kwargs['vault_cluster_config'].get(cluster_name, 'devc')
+    secret_env_vars = {k: v for k, v in environment.items() if is_secret_ref(v)}
+    if secret_env_vars:
+        secret_provider = get_secret_provider(secret_provider_name, soa_dir, service_name, ecosystem)
+        secret_environment = secret_provider.decrypt_environment(
+            secret_env_vars,
+            **secret_provider_kwargs,
+        )
+    environment.update(secret_environment)
```

Comment:
Can you instead just return variables and make the caller update?

Reply:
That sounds, like a good shout, let me look at this again